### PR TITLE
Try parallel testing and caching

### PR DESCRIPTION
**Pull request by higs4281**
In hopes of speeding up test runs, this takes a shot at running parallel
tests, caching pip installs between runs and combining the coverage
reports. If this doesn't help, we could try caching specific directories.

This also fixes branch specifiers for the build and code climate badges, 
which were still pointed at the  flapjack branch.

## Changes

- New travis configs
- README badges

- @chosak @sebworks

### CODE REVIEW

**Comment by higs4281** [comment]
So this experiment mostly failed. Parallel tests, at least as Travis handles them for our suite of gulp and tox tests, become a demolition derby of processes, each racing to see who can run into a missing dependency first.

I did add a command to combine coverage results into one figure, but the search for parallel tests with separate results continues.

**Comment by rosskarchner** [comment]
From my reading of [this](https://docs.travis-ci.com/user/speeding-up-the-build/)It seems like the way to do this would be to have a (sorry @willbarton @grapesmoker ) shell script that runs one set of tests or the other based on an environment variable, and then have two 'env' entries in the travis.yml, like:

```yaml

env:
- RUNTEST=frontend
- RUNTEST=backend
```

Those will each kick off entirely seperate travis builds

**Comment by higs4281** [comment]
@rosskarchner thanks -- I'll pursue that.

File: .travis.yml [hunk 1]
```diff
@@ -3,6 +3,8 @@ language: python
 python:
   - "2.7"
 
+cache: pip
+
```
Comment (willbarton) [inline .travis.yml]: I wonder if it's worth including the `node_modules` directory as @chosak did in #2381.

Comment (higs4281) [inline .travis.yml]: Since it didn't speed anything up, I would leave it in 2381 until we know why.

**Pull request merged**